cucumber/authenticators_azure/features/step_definitions/authn_azure_steps.rb

```diff
@@ -0,0 +1,31 @@
+Given(/^I successfully set Azure variables$/) do
+  set_azure_provider_uri_variable
+end
+
+Given(/^I successfully set Azure provider-uri variable to value "([^"]*)"$/) do |provider_uri|
+  set_azure_provider_uri_variable(provider_uri)
+end
+
+Given(/^I successfully set Azure provider-uri variable without trailing slash$/) do
+  set_azure_provider_uri_variable(azure_provider_uri.chop)
+end
+
+Given(/I fetch an Azure access token inside machine/) do
```

Comment:
this is only for a system-assigned-id, otherwise for user-assigned-id, you need the client id

Reply:
I would create two separate functions 
```
retrieve_azure_access_token_user_assigned_id (client_id)
       command = "curl 'http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&client_id=?resource=https%3A%2F%2Fmanagement.azure.com%2F' -H Metadata:true -s | jq -r '.access_token'"

retrieve_azure_access_token_system_assigned_id
```

Reply:
I started with the user assigned id which is the basic one and plan to get the system assigned id code once i test it. I don't want to add helpers that are not used in this PR to make it less noisy.

Reply:
But that's the thing it shouldn't work if you use user-assigned-id b/c your endpoint is:
```
http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&resource=https%3A%2F%2Fmanagement.azure.com%2F
```
So you are missing the client-id in your request. For user-assigned-identity, it must have the client-id. Like so:
```
http://169.254.169.254/metadata/identity/oauth2/token?api-version=2018-02-01&client_id=?resource=https%3A%2F%2Fmanagement.azure.com%2F
```

Reply:
it is not a user-assigned identity token. It is a system-assigend one, only i am not using the identity itself (the oid). Once we will add the user-assigned id test I will add another method for that.

Reply:
So let's not use "Users" in the Scenario for now?
```
  Scenario: Users can authenticate with Azure authenticator and fetch secret
```
Maybe, Machines/Azure instances

Reply:
I will change to hosts but it will always be host. We will not authenticate users, regardless of the token type.